public/scripts/extensions/expressions/index.js

```diff
@@ -208,13 +206,15 @@ async function visualNovelSetCharacterSprites(container, name, expression) {
                 template.fadeIn(250, () => resolve());
             });
             createCharacterPromises.push(fadeInPromise);
-            const setSpritePromise = setLastMessageSprite(template.find('img'), avatar, labels);
+
+            const characterId = context.characters.findIndex(x => x == character);
+            const setSpritePromise = await setLastMessageSprite(template.find('img'), avatar, labels, characterId)
```

Comment:
This is rather the strange part. From what I could see when debugging this was that if there was no await, even with the fix, character sprites on a new chat/existing chat will still result the same bug albeit it will now apply the same expression of the first character message to all sprites instead. 

Adding the await here, whilst strange, was the only current way I could do in which expressions wouldn't result in sprites sharing the same emotion at the start/opening of a chat.

Reply:
Think new code now should have resolved this.